### PR TITLE
Re-enable "Check Scripts" as optional part of make lint

### LINKED ISSUE #7861: Check Script task is failing
It looks like Check Scripts was made mandatory recently. It seems to be conflicting with some recent check-ins and blocking checkins.

https://circleci.com/gh/istio/istio/155389?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

I am going to disable the CircleCI task for Check Scripts as it is ancillary to product health at this point. Also it needs to run stably at least for a week before it can be made a required gate.

### PR DESCRIPTION
**Pull request by Tahler**
Resolves #7861.

This PR fixes the force-merged ShellCheck violations and re-enables the "Check Scripts" task as disabled by #7862.

It also addresses the array-building issues discovered in #7880.

/cc @gargnupur @ozevren @rkpagadala @mandarjog @kyessenov 

Edit: No longer fixes the new violations.

### CODE REVIEW

**Comment by costinm** [comment]
I'm not sure the validations are valid - from what I've seen it forces a lot of quotes that are not needed.

It can certainly not be made required until it is reviewed and approved, and I'm not sure it needs to be a pre-submit check.

**Comment by costinm** [comment]
Also - please send an email to istio-dev before making any changes to the pre-submit infrastructure.

**Comment by kyessenov** [comment]
LGTM

This is just a deterministic linter. There is nothing flaky about it. I don't think it's necessary to wait for a week of green tests.

**Comment by ozevren** [comment]
Can this simply be part of "make lint"? It would be nice to have a simple command-line experience for the developers that align with how we tests things in the check-in system. 

I've been using "make lint" to do lint checks, and it mostly works (well, except for depend diff check, which I think should be part of make lint as well, but that is a separate story).

**Comment by ozevren** [comment]
I am not familiar with most of the actual scripts, so I'll refrain from lgtming and let other people who are more familiar take a closer look. I assume you tried running these scripts to see if they work as intended?

I think it looks good overall. However, splitting this into multiple PRs and landing them over a course of days may help track any issues that might arise. It will also reduce the blast radius in case these create any issues (and also reduce blast radius, in case any of your checkins got reverted).

**Comment by Tahler** [comment]
That's fair. I'm definitely making this PR too broad now.

I'll split this into:

1. Re-enable it as optional / always passing
2. Add it to `make lint`
3. Fix the violations
4. Fix the typo

File: tools/run_shellcheck.sh [hunk 1]
```diff
@@ -2,6 +2,12 @@
 #
 # Runs shellcheck on all shell scripts in the istio repository.
 
+if ! command -v shellcheck > /dev/null; then
```
[the remark this answers is not in the document]
Comment (Tahler) [reply tools/run_shellcheck.sh]: Sounds good. This line is for those calling `make lint` from their local machine, while CircleCI uses the lint image that I added in #7785.

**Comment by Tahler** [comment]
FYI I dropped the shell script fixes from this PR and will move them to another PR after this one lands.

File: tools/run_shellcheck.sh [hunk 2]
```diff
@@ -25,5 +25,6 @@ SHEBANG_FILES=$( \
 # "--excludes=SC1090,SC1091"). See https://github.com/koalaman/shellcheck/wiki
 # for details on each code's corresponding rule.
 
+# Remove `|| true` to make this fail in CircleCI.
```
[the remark this answers is not in the document]
Comment (Tahler) [reply tools/run_shellcheck.sh]: Done.

**Comment by mandarjog** [comment]
The shellCheck command will always pass after printing out violations.

**Comment by Tahler** [comment]
I agree with Costin that if it is going to be optional, it should be a separate job so that it encourages a click to see the errors. Unfortunately, that means it can't be part of `make lint` (except by also changing the current lint job). It also means new PRs are going to see errors irrelevant to their PR...

**Pull request merged**